Raven.Voron/Voron/StorageEnvironment.cs

```diff
@@ -439,7 +439,7 @@ public Transaction NewTransaction(TransactionFlags flags, TimeSpan? timeout = nu
                 _txCommit.EnterReadLock();
                 try
                 {
-                    long txId = flags == TransactionFlags.ReadWrite ? _transactionsCounter + 1 : _transactionsCounter;
+                    long txId = flags == TransactionFlags.ReadWrite ? NextWriteTransactionId : CurrentReadTransactionId;
```

Comment:
Have you had a problem because of that? Looks like relevant for 3.0, 4.0 as well, is it?